staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -143,9 +174,18 @@ func WithPriorityAndFairness(
 				epmetrics.DroppedRequests.WithContext(ctx).WithLabelValues(epmetrics.ReadOnlyKind).Inc()
 			}
 			epmetrics.RecordRequestTermination(r, requestInfo, epmetrics.APIServerComponent, http.StatusTooManyRequests)
+			if isWatchRequest {
+				close(resultCh)
+			}
 			tooManyRequests(r, w)
 		}
-
+		// In case of watch, from P&F POV it already finished, but we need to wait until the request itself finishes.
+		if isWatchRequest {
+			err := <-resultCh
```

Comment:
can we update the comment above? if the request gets rejected then `resultCh` will be closed and `err` will be `nil` as a consequence. (just to call it out that `err` being `nil` does not always mean that the request has been executed)  

so basically we are waiting here until `the request itself finishes or is rejected`